### DISCUSSION TITLE
Subscribing to security advisory notifications

### DISCUSSION THREAD

**Discussion by Yoyoda75** [community/community#51573] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hi everyone, I would like to subscribe to security advisory notifications, let's say from [this repository](https://github.com/nextcloud/security-advisories/security/advisories). I have checked the `Security alerts` box in `Watch/custom` but I'm not receiving any notification when an adivsory is published. 

My question is: does `security alert` correspond to security advisories or is it a different section ? 

Thank you

**Comment by Kaparrs** [comment]
When you check the box to enable Security alerts for a repository, you should receive notifications whenever a new security advisory is published for that repository.

However, there are a few things you may want to check to ensure that you are properly set up to receive these notifications:

1. Make sure you are watching the repository: You mentioned that you checked the "Watch" option, but it's possible that you may not be watching the repository for which you want to receive security advisories. To check this, go to the repository page, click the "Watch" button, and make sure "Watching" is selected.

2. Check your notification settings: Even if you have enabled Security alerts and are watching the repository, you may not be receiving notifications if your notification settings are not configured correctly. To check your notification settings, click on your profile picture in the top right corner of the GitHub page, then click "Settings". From there, click "Notifications" and make sure that you have enabled email or web notifications for security advisories.

3. Verify that there are security advisories for the repository: It's possible that the repository you are watching may not have any published security advisories yet. To check this, go to the "Security" tab of the repository and see if any advisories are listed.

**Comment by likeshumidity** [comment]
re: '... click "Notifications" and make sure that you have enabled email or web notifications for security advisories.'

<img width="758" alt="image" src="https://user-images.githubusercontent.com/8808703/229537982-da06355b-0111-4d9b-b5c4-043c6df6589b.png">

There appears to be nothing related to security advisories for a repo only security advisories for dependencies. It would be super helpful if this included Security Advisories for the repo itself.

**Reply by Kaparrs** [reply]
I'm confused what you mean by repository security advisories. Security advisories are related to the dependencies inside the repository

**Reply by likeshumidity** [reply]
### Repo A
  - Security Advisories
    - Security Advisory 1
  - Dependencies
    - Repo B

### Repo B
  - Security Advisories
    - Security Advisory 2

If I subscribe to security notifications for Repo A, I would get notified about Security Advisory 2, because Repo B is a dependency of Repo A, but I would not get notified about Security Advisory 1.

I want to be notified about Security Advisory 1 and 2 if I am subscribed to security notifications for Repo A.

Does that clarify?

**Reply by Kaparrs** [reply]
Yes, not sure if that's changeable

**Comment by ningziwen** [comment]
+1 Suggest supporting notifying the security advisory of current repo. ("security advisory 1" in the response above)

**Comment by t80027t** [comment]
Same problem for me - i'm not getting email notifications on newly published Security Advisories when i am watching for 'Security Alerts' only. I've tripled checked all settings.

If i change Watch settings on the repo from 'Custom' on Security Alerts to 'All Activity', then i get emails on newly published Security Advisories (but i don't want emails on everything else!)

**Comment by linwalth** [comment]
Our team wanted to get notified only when a security advisory was published on a repository we are watching.

It would be really useful if this actually worked, especially since the state of it not working is rather unintuitive.

Being able to customize when to get notifications and then not actually getting notified was confusing to our team. As I am sure it was - and is - to others.

We would appreciate this be fixed 🙂

**Comment by karankohli-cf** [comment]
any update on this issue?

**Comment by lcostea** [comment]
I can confirm it is still not working if you select Custom > Security Alerts, it only works if you select to receive all, then the security advisories will come. So nothing done yet

**Comment by haubourg** [comment]
Same here, we activated these reports on a very large open repo with thousands of interactions per day. There is no way we can subscribe to all notifications just to get the security notification reports. Any idea is this will move. If not, we will have to deal without this option.   I fail to see how one can imagine that a team of triager should scan all the activity content to find security reports..

**Comment by 3nids** [comment]
Has this been implemented?, I have this
![image](https://github.com/user-attachments/assets/e1f9ca38-7e4b-4048-b602-fde2214ac3e9)

**Comment by ghbob** [comment]
If I check "Releases" I will get notifications about releases.
If I check "Security alerts" I won't get nofications about security alerts.

"unintuitive" would be a nice way of putting it ;)


From GitHub support today:
> You are correct, security notifications would be triggered by a Dependabot alert, so you would need to have a repository that lists the impacted package as a dependency in order to receive them.


So yes, please, make them work like the other notifications.

**Comment by Harrysharmax** [comment]
How to Receive Security Advisory Notifications
While GitHub does not have a dedicated "subscribe to new advisories" button for a single repository, here are the best ways to stay informed:

Watch "All Activity": This is the most direct method within GitHub. Go to the repository, click the Watch button, and select "All Activity." You will be notified of all conversations, including new releases, issues, and published security advisories. The downside is that this can be noisy for very active repositories.

Check the Repository's Security Policy: Look for a SECURITY.md file in the repository. Maintainers often use this file to detail how they handle security issues and how they disclose vulnerabilities. They may have a dedicated mailing list or another channel for security announcements.

Monitor the GitHub Advisory Database: All published advisories are collected in the [GitHub Advisory Database](https://github.com/advisories). You can visit this database to browse and search for advisories across all of GitHub.

**Comment by domstubbs** [comment]
What is the point of being able to explicitly browse to Watch → Custom → Security Alerts if not to be notified whenever the repo maintainer authors a new security advisory? To say this is misleading is an understatement.

**Comment by halfgaar** [comment]
I ran into the same. I'm "watching" security advisories, to find out various projects have new security advisories published that I didn't get notified of. 

It's simply a bug that you have a GUI element that says you can watch security advisories, but you aren't actually watching them.
